src/internal/oscal/generate_utils.go

```diff
@@ -0,0 +1,360 @@
+		--tags json,yaml // the tags to add to the Go structs \
+		--pkg <name_of_your_go_package> // defaults to "main"
+
+	For more information on how to use go-oscal: go-oscal --help
+
+	Source: https://github.com/defenseunicorns/go-oscal
+	*/`
+
+// commonInitialisms is a set of common initialisms.
+// Only add entries that are highly unlikely to be non-initialisms.
+// For instance, "ID" is fine (Freudian code is rare), but "AND" is not.
+var commonInitialisms = map[string]bool{
+	"API":   true,
+	"ASCII": true,
+	"CPU":   true,
+	"CSS":   true,
+	"DNS":   true,
+	"EOF":   true,
+	"GUID":  true,
+	"HTML":  true,
+	"HTTP":  true,
+	"HTTPS": true,
+	"ID":    true,
+	"IP":    true,
+	"JSON":  true,
+	"LHS":   true,
+	"QPS":   true,
+	"RAM":   true,
+	"RHS":   true,
+	"RPC":   true,
+	"SLA":   true,
+	"SMTP":  true,
+	"SSH":   true,
+	"TLS":   true,
+	"TTL":   true,
+	"UI":    true,
+	"UID":   true,
+	"UUID":  true,
+	"URI":   true,
+	"URL":   true,
+	"UTF8":  true,
+	"VM":    true,
+	"XML":   true,
+	"NTP":   true,
+	"DB":    true,
+}
+
+var intToWordMap = []string{
+	"zero",
+	"one",
+	"two",
+	"three",
+	"four",
+	"five",
+	"six",
+	"seven",
+	"eight",
+	"nine",
+}
+
+// buildStructs builds the structs for each definition in the schema
+func buildTagString(tags []string, field string, required bool) string {
+	tagStrings := []string{}
+	tagSuffix := ",omitempty"
+
+	// No tags, return empty string
+	if len(tags) == 0 {
+		return ""
+	}
+
+	// If required, remove omitempty
+	if required {
+		tagSuffix = ""
+	}
+
+	// Build tag string for each tag
+	for _, tag := range tags {
+		tagStrings = append(tagStrings, fmt.Sprintf("%s:\"%s%s\"", tag, field, tagSuffix))
+	}
+	return "`" + strings.Join(tagStrings, " ") + "`"
+}
+
+// getRef builds a ref string from a schema
+func getRef(schema jsonschema.Schema) (string, error) {
+	if schema.Ref != nil {
+		return *schema.Ref, nil
+	} else if schema.ID != nil {
+		return *schema.ID, nil
+	} else if schema.Title != nil {
+		return getRefWithName(getNameFromTitle(*schema.Title)), nil
+	}
+	return "", fmt.Errorf("unable to get ref from schema")
+}
+
+func getNameFromTitle(title string) string {
+	split := strings.Split(title, " ")
+	name := ""
+	for _, s := range split {
+		name += FmtFieldName(s)
+	}
+	return name
+}
+
+func getRefWithName(name string) string {
+	return "#/definitions/" + name
+}
+
+// returns the json type of the schema
+func getJsonType(schema jsonschema.Schema) string {
+	if schema.Type != nil {
+		return string(*schema.Type.SimpleTypes)
+	}
+	return ""
+}
+
+// isPrimitiveJsonType returns true if the type is a primitive type
+func isPrimitiveJsonType(t string) bool {
+	lower := strings.ToLower(t)
+	switch lower {
+	case "string":
+		return true
+	case "boolean":
+		return true
+	case "number":
+		return true
+	case "integer":
+		return true
+	}
+	return false
+}
+
+// getGoType returns the Go type for a given JSON type
+func getGoType(t string) string {
```

Comment:
[question] Could we also evaluate `format` in the schema for potentially augmenting our go types with additional types? 

take `DateTimeWithTimezoneDatatype` for instance - type string but format `date-time`. Just curious on your thoughts here.

Reply:
Noting this would add some required overhead for go imports included in the types file generation

Reply:
I think it is possible, especially since there is a defined pattern in the schema required for all date-time. I would have to look into it. I think it would work potentially with yaml as it supports more types, but I think I would need to add a "middleware" for unmarshall and that may require any consumer to import our unmarshalling method or add the middleware themselves from go-oscal. (all assumption)

Created a story to investigate
https://github.com/defenseunicorns/go-oscal/issues/122